virtualmachine/vsphere/util.go

```diff
@@ -594,6 +594,51 @@ func removeExistingNetworks(vm *VM, vmObj *object.VirtualMachine) ([]types.BaseV
 	return removeSpecs, nil
 }
 
+func FindByVirtualDeviceFileName(disks []Disk, name string) *Disk {
+	for _, disk := range disks {
+		if disk.DiskName == name {
+			return &disk
+		}
+	}
+	return nil
+}
+
+func resizeAndDeleteVols(vmMo mo.VirtualMachine, disks []Disk) ([]types.BaseVirtualDeviceConfigSpec, error) {
+	var deviceSpecs []types.BaseVirtualDeviceConfigSpec
+	devices := object.VirtualDeviceList(vmMo.Config.Hardware.Device)
+	for _, device := range devices {
+		if editdisk, ok := device.(*types.VirtualDisk); ok {
+			backing := editdisk.Backing
+			fileBackingInfo := backing.(types.BaseVirtualDeviceFileBackingInfo).GetVirtualDeviceFileBackingInfo()
+			disk := FindByVirtualDeviceFileName(disks, fileBackingInfo.FileName)
+			var dvconfig types.BaseVirtualDeviceConfigSpec
+			if disk == nil {
+				// if user wants to delete the disk
+				dvconfig = &types.VirtualDeviceConfigSpec{
+					Operation: types.VirtualDeviceConfigSpecOperationRemove,
+					Device:    editdisk,
```

Comment:
please verify if FileOperation is needed here, as this is a remove operation.

Reply:
i will discuss this with others as well..do we need to make this or not ?

Reply:
discuss offline for now no need to do this.